assembly/src/header.ts

```diff
@@ -1,11 +1,18 @@
 import { log } from './util/log';
 import { exit } from './util/exit';
 
-export function validateHeader(buffer: Uint8Array, position: u32): u32 {
-  const HEADER: StaticArray<u8> = [137,  80,  78,  71,  13,  10,  26,  10] as StaticArray<u8>;
-  const HEADER_BITS = 8;
+const HEADER = [137,  80,  78,  71,  13,  10,  26,  10] as StaticArray<u8>;
+const HEADER_BITS = 8;
+
+export function validateHeader<T>(buffer: T, position: u32): u32 {
   for (let i = 0; i < HEADER_BITS; i++) {
-    if (unchecked(buffer[i]) != unchecked(HEADER[i])) {
+    // @ts-ignore: safe definition checked
```

Comment:
actually according to #1 we could do
```ts
const HEADER: u64 = 0xA1AAD474E5089;
if (load<u64>(buffer.dataStart) != HEADER) {
    ...
}
```

did you leave it for me by any chance intentionally? :)

Reply:
If `buffer` is `Uint8Array` you can use `memory.compare()`, but because we don't know if `T` has that property, we must rely on the operator implementation.